test/grains_cases_test.rb

```diff
@@ -0,0 +1,87 @@
+require_relative 'test_helper'
+
+class GrainsTest < Minitest::Test
+  def test_test_name
+    test_case = GrainsCase.new(description: 'description')
+
+    assert_equal 'test_description', test_case.test_name
+  end
+
+  def test_test_name_when_description_has_spaces
+    test_case = GrainsCase.new(description: 'description with spaces')
+
+    assert_equal 'test_description_with_spaces', test_case.test_name
+  end
+
+  def test_workload_is_not_implemented
+    assert_raises(NotImplementedError) { GrainsCase.new.workload }
+  end
+
+  def test_skipped_with_zero_index
+    assert_equal '# skip', GrainsCase.new(index: 0).skipped
+  end
+
+  def test_skipped_with_nil_index
```

Comment:
This is a little bit of a tricky one that I'm not sure on the best way to handle.

In my (limited) experience, when the canonical data has more than one "type" of test case, they both each follow the same JSON structure (e.g. [allergies](https://github.com/exercism/x-common/blob/master/exercises/allergies/canonical-data.json)). This lets you do something like this when assembling the cases (not an optimal solution necessarily, just demonstrating):

```ruby
i = -1

JSON.parse(data).flat_map do |type, test_data|
  test_data['cases'].map do |test_case|
    i += 1
    # Determine which kind of Case to make based on type
    Case.new(...) 
  end
end
```

But in this case the [JSON](https://github.com/exercism/x-common/blob/master/exercises/grains/canonical-data.json) is structured like:

```json
{
  "square": {
    "cases": [
      { "description": "...", "input": 1, "expected": 1 },
      ...
    ]
  },
  "total": {
    "description": "...",
    "expected": "..."
  }
}
```

Because of this, I'm not too sure how to get a real index for this case.

Do you have any thoughts on this? Would it be better to explicitly pass `index: false` for that case? It's still pretty strange though 😞